File: SakhCubaAPI/ClientApp/src/assets/application.ts
export class Application {
    constructor(public nickname:string, 
      public discordNickname:string,
      public about:string)
      {}
  }

export class AdminApplication extends Application {
  constructor(public id: number,
    public nickname:string, 
    public discordNickname:string,
    public about:string,
    public ip: string,
    public date: Date,
    public decision: string) {
    super(nickname, discordNickname, about);
  }
}

export class AdminApplicationView extends AdminApplication {
  constructor(public id: number,
    public nickname:string,
    public discordNickname:string,
    public about:string,
    public ip: string,
    public date: Date,
    public decision: string,
    public decisionId: number){
    super(id, nickname, discordNickname, about, ip, date, decision)
  }
}

/*
1 - новая
2 - ок
3 - ?
4 - не ок
*/

File: Entities/Models/Application.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Entities.Models
{
    public class Application
    {
        public int Id { get; set; }

        [StringLength(100, ErrorMessage = "Ник не может быть длиннее 100 символов")]
        public string Nickname { get; set; } = "";

        [StringLength(50, ErrorMessage = "Дискорд не может быть длиннее 50 символов")]
        public string DiscordNickname { get; set; } = "";

        [StringLength(2000, ErrorMessage = "Не больше 2000 символов в графе О себе")]
        public string About { get; set; } = "";
        public string Ip { get; set; } = "";
        [DataType(DataType.Date)]
        public DateTime Date { get; set; }

        [ForeignKey(nameof(Decision))]
        public int DecisionId { get; set; }
        public Decision Decision { get; set; }
    }
}
